app/utils/callHandler.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+/**
+ * A utility wrapper function to replace numerous variations of `mockRequest`
+ * around the tests. This utility function accepts a Next.js request handler
+ * and (using type inference) provides a type-safe way to call the handler in
+ * tests with type-hinted parameters.
+ *
+ * @example
+ * ```ts
+ *   import { GET } from './route.ts'
+ *   // ...
+ *   const response = await requestWrapper(GET, { product: 'terraform' })
+ *
+ *   expect(response.status).toBe(200)
+ *   // Other assertions...
+ * ```
+ */
+export const callHandler = <
```

Comment:
@rmainwork let's give this a more description name, so folks know that this is used on;y in testing.

Something like `testRequestHandler` or `createTestRequestHandler`.

Reply:
Agreed. I can't say I was crazy about the name, I just picked it to name it something. I ended up going with `mockRequest` in 1a37a8033992f3547bfc6edac1e2171b5119dae8. My reasoning was that it's the same name as the previous implementation, so it should hopefully nudge people toward using that instead of creating their own `mockRequest` functions...

I didn't end up going with `testRequestHandler` or `createTestRequestHandler` as they don't exactly test the handler themselves or create the handler which those names seem to imply.

I honestly don't _love_ the name `mockRequest` as it implies (at least to me) that the function somehow mocks part of the request stack out(I suppose it _sort_ of does but ehhhh) rather than just calls the handler provided in a predetermined way.